src/terms/xc.jl

```diff
@@ -184,9 +184,13 @@ function DensityDerivatives(basis, max_derivative::Integer, ρ::RealFourierArray
     model = basis.model
     @assert model.spin_polarization == :none "Only spin_polarization == :none implemented."
     function ifft(x)
-        tmp = G_to_r(basis, x)
-        check_real(tmp)
-        real(tmp)
+        # For even-length grids the element with largest negative integer coordinate
+        # in G_vectors has no complex-conjugated partner. In order to avoid having
+        # a spurious imaginary component in real-space, we set these elements to zero
+        iseven(basis.fft_size[1]) && (x[div(end, 2) + 1, :, :] .= 0)
+        iseven(basis.fft_size[2]) && (x[:, div(end, 2) + 1, :] .= 0)
+        iseven(basis.fft_size[3]) && (x[:, :, div(end, 2) + 1] .= 0)
+        check_real(G_to_r(basis, x))
```

Comment:
You removed too much here (hopefully the tests will catch it)

Reply:
Apparently not. But also I'm not sure what you mean. If `fft_size` is `12, y, z` then `(-6, n, m)` has no counterpart since `(6, -n, -m)` is not part of the grid, no matter what `n` and `m` are.

Reply:
I mean you removed the actual computation of the IFFT

Reply:
That's in line 193, also `check_real(x)` returns `real(x)`

Reply:
Ah, sorry!

Reply:
That logic (with the iseven) should go elsewhere. It's weird to "purify" only here: we should either do it everywhere (eg in density building) or nowhere at all. Maybe just ignore these coefficients in check_real?

Reply:
I thought about putting the purification into check_real, but than it's really strange that a check function modifies. Maybe we can just ignore these elements. That would make sense, I guess.

Reply:
Not as easy as I thought because check_real takes a real-space array only, but the zeroing is to be done in fourier space. A made a special function for it now.